tests/integration/worker/test_submission_worker.py

```diff
@@ -0,0 +1,394 @@
+                name='Challenge Phase',
+                description='Description for Challenge Phase',
+                leaderboard_public=False,
+                is_public=True,
+                start_date=timezone.now(),
+                end_date=timezone.now() + timedelta(days=1),
+                challenge=self.challenge,
+                test_annotation=SimpleUploadedFile('test_sample_file.txt',
+                                                   b'Dummy file content',
+                                                   content_type='text/plain')
+            )
+        self.url = reverse_lazy('jobs:challenge_submission',
+                                kwargs={'challenge_id': self.challenge.pk,
+                                        'challenge_phase_id': self.challenge_phase.pk})
+
+        self.submission = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.challenge_host_team.created_by,
+            status="submitted",
+            input_file=SimpleUploadedFile(
+                'user_annotation_file.txt',
+                b"Dummy file content",
+                content_type='text/plain'
+            ),
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+            is_public=True,
+        )
+
+    def tearDown(self):
+        try:
+            shutil.rmtree('/tmp/evalai')
+        except OSError:
+            pass
+
+
+class ProcessSubmissionCallbackTestClass(BaseTestClass):
+    @mock.patch("scripts.workers.submission_worker.process_submission_message")
+    def test_process_submission_callback(self, mock_psm):
+        message = {
+            "challenge_pk": self.challenge.pk,
+            "phase_pk": self.challenge_phase.pk,
+            "submission_pk": self.submission.id
+        }
+        body = json.dumps(message)
+
+        submission_worker.process_submission_callback(body)
+        mock_psm.assert_called_with(message)
+
+    @mock.patch("scripts.workers.submission_worker.logger.exception")
+    @mock.patch("scripts.workers.submission_worker.process_submission_message")
+    def test_process_submission_callback_unsuccesful(self, mock_psm, mock_logger):
+        message = {
+            "challenge_pk": self.challenge.pk,
+            "phase_pk": self.challenge_phase.pk,
+            "submission_pk": self.submission.id
+        }
+        body = json.dumps(message)
+
+        mock_psm.side_effect = Exception("test error")
+
+        submission_worker.process_submission_callback(body)
+
+        mock_logger.assert_called_with("Exception while receiving message from submission queue with error test error")
+
+
+class ProcessSubmissionMessageTestClass(BaseTestClass):
+    @mock.patch("scripts.workers.submission_worker.SUBMISSION_DATA_DIR", "mocked/dir/submission_{submission_id}")
```

Comment:
`"mocked/dir/submission_{submission_id}")`-> `"mocked/dir/submission_{submission_pk}")`

Reply:
That's because in submission worker 'id' is used in a lot of places for string formatting. So, I'll change everything to pk even in the submission worker? @RishabhJain2018

Reply:
No, please don't change there. let's make the change here only. We can do that gradually when we will edit those files.